### PR TITLE
pcb2gcode: init at 2.1.0

### PR DESCRIPTION
**Pull request by Kritnich**
###### Motivation for this change
pcb2gcode is a command-line software for the isolation, routing and drilling of PCBs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@Kritnich Can you solve the merge conflict?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24847,4 +24847,5 @@ in
 
   wifi-password = callPackage ../os-specific/darwin/wifi-password {};
 
+  pcb2gcode = callPackage ../tools/misc/pcb2gcode {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24850]: Please sort this alphabetically to avoid merge conflicts.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Can you please squash those commit together?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101976` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pcb2gcode</li>
  </ul>
</details>

**Comment by Kritnich** [comment]
> Result of `nixpkgs-review pr 101976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:
> 
>     * pcb2gcode

I ran that as well and it built on x86_64-linux. Do you have any more details on why it failed?

**Comment by SuperSandro2000** [comment]
```
Can't exec "/bin/sh": No such file or directory at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/share/autoconf/Autom4te/General.pm line 307.
Use of uninitialized value $Autom4te::General::tmp in scalar chomp at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/share/autoconf/Autom4te/General.pm line 309.
Can't exec "/bin/sh": No such file or directory at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf line 188.
Can't exec "/bin/sh": No such file or directory at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf line 189.
Use of uninitialized value $aclocal_help in pattern match (m//) at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf line 190.
Use of uninitialized value $aclocal_help in pattern match (m//) at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf line 191.
Use of uninitialized value $automake_help in pattern match (m//) at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf line 192.
Use of uninitialized value $automake_help in pattern match (m//) at /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf line 193.
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal -I m4 --output=aclocal.m4t
Can't exec "/bin/sh": No such file or directory at /nix/store/38vxlxrvg3yji3jms44qn94lxdysbj5j-perl-5.32.0/lib/perl5/5.32.0/x86_64-linux-thread-multi/IO/File.pm line 186.
aclocal: error: cannot open echo 'm4_define([m4_require_silent_probe], [-])' | /nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autom4te --language Autoconf-without-aclocal-m4  - /nix/store/ldr8z2
```

**Comment by Kritnich** [comment]
I have been unable to reproduce the issue on two different machines, but I assume it has something to do with bash and a missing symlink to it on `/bin/sh`. If it still doesn't work, do you have any pointers on how I can reproduce the issue?

**Comment by SuperSandro2000** [comment]
I opened ``/nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/share/autoconf/Autom4te/General.pm +307`` locally:

image

and the error message might be miss leading. It could be that it tries to access ``/tmp`` which is disabled in sandbox mode I believe. I think setting $TMPDIR correct could solve the issue. Or it is the `` ` `` but then I ask myself how any perl code works.

Lastly it might also be a quirk with my machines I did not discover yet.

**Comment by Kritnich** [comment]
TMPDIR _should_ be set correctly by `mkDerivation` whether it's sandboxed or not (to `/tmp` or `/build`), if I understood it correctly. 

Besides that, both machines I tested it on were in sandbox mode and in `/nix/store/r8b9z1s0di4c8qnhwrpcir7p7rm7s4x7-autoconf-2.69/bin/autoreconf` at L188,L189 there's the same issue with backticks present but no `/tmp` being used. 

Definitely seems to me like Perl is trying to call `/bin/sh -c` but fails because it's missing, which is odd since I feel like that should break way more. 

Either way, I included bash in that last push. Can you give it a shot and see if it builds on your machine now?

**Comment by fooker** [comment]
I've tested it with the latest commit and it worked well. Both, building and executing the resulting command.

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "pcb2gcode";
+    repo = "pcb2gcode";
+    rev = "v${version}";
+    sha256 = "0nzglcyh6ban27cc73j4l7w7r9k38qivq0jz8iwnci02pfalw4ry";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig bash ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:23]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config bash ];
```

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  meta = with stdenv.lib; {
+    description = "Command-line tool for isolation, routing and drilling of PCBs ";
+    longDescription = ''
+      pcb2gcode is a command-line software for the isolation, routing and drilling of PCBs.
+      It takes Gerber files as input and it outputs gcode files, suitable for the milling of PCBs.
+      It also includes an Autoleveller, useful for the automatic dynamic calibration of the milling depth.
+    '';
+    homepage = "https://github.com/pcb2gcode/pcb2gcode";
+    license = licenses.gpl3;
+    maintainers = [ maintainers.kritnich ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:41]:
```suggestion
    maintainers = with maintainers; [ kritnich ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pcb2gcode</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:28369]: @Kritnich Please address this after which I will be merging it.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pcb2gcode</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@Kritnich please address https://github.com/NixOS/nixpkgs/pull/101976#discussion_r530726258

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    repo = "pcb2gcode";
+    rev = "v${version}";
+    sha256 = "0nzglcyh6ban27cc73j4l7w7r9k38qivq0jz8iwnci02pfalw4ry";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config bash ];
+
+  buildInputs = [ boost glibmm gtkmm2 gerbv librsvg ];
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:27]:
```suggestion
  postPatch = ''
```

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  meta = with stdenv.lib; {
+    description = "Command-line tool for isolation, routing and drilling of PCBs ";
+    longDescription = ''
+      pcb2gcode is a command-line software for the isolation, routing and drilling of PCBs.
+      It takes Gerber files as input and it outputs gcode files, suitable for the milling of PCBs.
+      It also includes an Autoleveller, useful for the automatic dynamic calibration of the milling depth.
+    '';
+    homepage = "https://github.com/pcb2gcode/pcb2gcode";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:40]: `gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pcb2gcode</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -2348,6 +2348,8 @@ in
 
   pbzx = callPackage ../tools/compression/pbzx { };
 
+  pcb2gcode = callPackage ../tools/misc/pcb2gcode {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2351]:
```suggestion
  pcb2gcode = callPackage ../tools/misc/pcb2gcode { };
```

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  nativeBuildInputs = [ autoreconfHook pkg-config bash ];
+
+  buildInputs = [ boost glibmm gtkmm2 gerbv librsvg ];
+
+  postPatch = ''
+    substituteInPlace ./Makefile.am \
+    --replace '`git describe --dirty --always --tags`' '${version}'
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:32]:
```suggestion
  meta = with lib; {
```

per recent contributing change.

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "pcb2gcode";
+    repo = "pcb2gcode";
+    rev = "v${version}";
+    sha256 = "0nzglcyh6ban27cc73j4l7w7r9k38qivq0jz8iwnci02pfalw4ry";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config bash ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:23]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

isn't bash part of the stdenv?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/pcb2gcode/default.nix (hunk 8 above)
Comment (Kritnich) [reply pkgs/tools/misc/pcb2gcode/default.nix:23]: It is, I added it as an impromptu fix for the issues you were having with building it before. Not sure if it had any effect, but I can remove it.

File: pkgs/tools/misc/pcb2gcode/default.nix [hunk 9]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchFromGitHub
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pcb2gcode/default.nix:1]:
```suggestion
{ stdenv
, lib
, fetchFromGitHub
```

**Pull request merged**